src/section_handlers/document_tracking.py

```diff
@@ -0,0 +1,85 @@
+        self.cvrf2csaf_name = cvrf2csaf_name
+        self.cvrf2csaf_version = cvrf2csaf_version
+
+    def _process_mandatory_elements(self, root_element):
+        self.json['id'] = root_element.Identification.ID.text
+        self.json['current_release_date'] = root_element.CurrentReleaseDate.text
+        self.json['initial_release_date'] = root_element.InitialReleaseDate.text
+        self.json['status'] = root_element.Status.text
+
+        revision_history, version = DocumentTracking._process_revision_history(root_element)
+        self.json['revision_history'] = revision_history
+        self.json['version'] = version
+
+        # Generator is set by this Converter
+        self.json['generator'] = {}
+        self.json['generator']['date'] = get_utc_timestamp()
+        self.json['generator']['engine'] = {}
+        self.json['generator']['engine']['name'] = self.cvrf2csaf_name
+        self.json['generator']['engine']['version'] = self.cvrf2csaf_version
+
+    """
+    Checks whether all version numbers in /document/tracking/revision_history match semantic versioning.
+    semantic version is defined in version_t definition
+    see: https://docs.oasis-open.org/csaf/csaf/v2.0/csd01/csaf-v2.0-csd01.html#3111-version-type and
+    section 9.1.5 Conformance Clause 5: CVRF CSAF converter
+    """
+    @staticmethod
+    def check_for_version_t(revision_history):
+        pattern = '^((0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)$'
+        return all([re.match(pattern, revision['number']) for revision in revision_history])
+
+    @staticmethod
+    def _process_revision_history(root_element):
+        # preprocess the data
+        revision_history = []
+        for revision in root_element.RevisionHistory.Revision:
+            revision_attrs = {}
+            revision_attrs['date'] = revision.Date.text
+            # keep original value in this variable for matching later
+            revision_attrs['number_cvrf'] = revision.Number.text
+            # this value might be overwritten later if some of the version number doesn't match semantic versioning
+            revision_attrs['number'] = revision.Number.text
+            revision_attrs['summary'] = revision.Description.text
+
+            revision_history.append(revision_attrs)
+
+        # handle corresponding part of Conformance Clause 5: CVRF CSAF converter
+        # that is: some of the version number in revision_history don't match semantic versioning
+        if not DocumentTracking.check_for_version_t(revision_history):
+            logging.info('Some of the version number /document/tracking/revision_history does not match semantic versioning. Reindexing to integers.')
+            # Sort revisions by version number
+            revision_history = sorted(revision_history, key=lambda x: list(map(int, x['number'].split('.'))))
```

Comment:
Here we happily exit when upstream no caller catches our raised `ValueError` in case some string that cannot be interpreted as a valid base 10 integer comes along. Is this OK?

Reply:
I believe this should be handled by validation of the input document, every `number` field should match this regex: `(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*)){0,3}`